concert/ext/tangoservers/walker.py

```diff
@@ -182,40 +183,54 @@ async def write_sequence(self, name):
             self.set_state(DevState.FAULT)
             raise
 
-
-
-
     async def consume(self, walker: DirectoryWalker) -> None:
+        """Defines an internal consumer coroutine to write incoming stream"""
         with PerformanceTracker() as pt:
             total_bytes = await walker.write(self._receiver.subscribe())
             pt.size = total_bytes * q.B
 
     @DebugIt()
     @command(
-        dtype_in=str,
-        doc_in="file name for the log file"
+        dtype_in=(str,),
+        dtype_out=str,
+        doc_in="log_name, log_level and file_name for logging",
+        doc_out="absolute log file path as an identifier for the logger"
     )
-    async def open_log_handler(self, file_name: str) -> None:
-        self._log_handler = logging.FileHandler(os.path.join(self._current, file_name))
-        self._logger.addHandler(self._log_handler)
+    async def register_logger(self, args: Tuple[str, str, str]) -> str:
+        logger_name, log_level, file_name = args[0], int(args[1]), args[2]
+        # NOTE: log_path functions as unique identifier for a logger instance as well as the
+        # absolute path for the log file, where logs should be written.
+        log_path: str = os.path.join(self._current, file_name)
+        handler = logging.FileHandler(log_path)
+        logger = logging.Logger(logger_name, log_level)
+        logger.addHandler(handler)
+        self._log_handlers[log_path] = handler
+        self._loggers[log_path] = logger
+        return log_path
 
     @DebugIt()
-    @command()
-    async def close_log_handler(self) -> None:
-        self._logger.removeHandler(self._log_handler)
-        self._log_handler.close()
+    @command(
+        dtype_in=str,
+        doc_in="unique identifier for logger"
+    )
+    async def deregister_logger(self, logger_id: str) -> None:
+        if logger_id in self._loggers and logger_id in self._log_handlers:
+            self._loggers[logger_id].removeHandler(self._log_handlers[logger_id])
+        self._log_handlers[logger_id].close()
```

Comment:
This will fail if `logger_id` test above fails, or? It should go "into" the if statement I think.